server/controllers/industry.js

```diff
@@ -141,8 +216,191 @@ const deleteJob = async (req, res) => {
+        }
+
+        const industryData = await retrieveOrCacheIndustry(req, user.userType.industryData);
+        if (!industryData) {
+            return res.status(404).json(generateRes(false, 404, "INDUSTRY_DATA_NOT_FOUND", {}));
+        }
+
+        const assessment = industryData.assessments.find(a => a.id === assessmentId);
+        if (assessment) {
+            return res.status(200).json(generateRes(true, 200, "ASSESSMENT_FOUND", { assessment: JSON.stringify(assessment) }));
+        }
+
+        return res.status(400).json(generateRes(false, 400, 'BAD_REQUEST', {}));
+    } catch (error) {
+        return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", { error }));
+    }
+};
+
+const createAssessment = async (req, res) => {
+    try {
+        const accessToken = req.header('Authorization').split(' ')[1];
+        const decodeAccessToken = jwt.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
+
+        //check if user exists
+        const user = await retrieveOrCacheUsers(req, decodeAccessToken.email); 
+
+        // check if user type is industry
+        if (user.userType.Type != process.env.INDUSTRY) {
+            return res.status(400).json(generateRes(false, 400, 'BAD_REQUEST', {}));
+        }
+
+        const assessmentName = req.body.name;
+        const assessmentQuestions = req.body.questions;
+
+        const industryData = (!user.userType.industryData)
+            ? new IndustryData({
+                jobs: {
+                    active: [],
+                    draft: [],
+                    archived: [],
+                },
+                assessments: [],
+            })
+            : await retrieveOrCacheIndustry(req, user.userType.industryData);
+
+        industryData.assessments.push({
+            name: assessmentName,
+            questions: assessmentQuestions,
+        });
+        await industryData.save();
+
+        if (!user.userType.industryData) {
+            user.userType.industryData = industryData._id;
+            await user.save();
+        }
+
+        return res.status(201).json(generateRes(true, 201, "ASSESSMENT_CREATED", {}));
+    } catch (error) {
+        return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", { error }));
+    }
+};
+
+const editAssessment = async (req, res) => {
+    try {
+        const accessToken = req.header('Authorization').split(' ')[1];
+        const decodeAccessToken = jwt.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
+
+        //check if user exists
+        const user = await retrieveOrCacheUsers(req, decodeAccessToken.email); 
+
+        // check if user type is industry
+        if (user.userType.Type != process.env.INDUSTRY) {
+            return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", {}));
+        }
+
+        const assessmentId = req.body.assessmentId;
+        const assessmentDetails = req.body.assessmentDetails;
```

Comment:
No validation or checks for the assesmentId and details. 
Consider adding either a schema in the Validation.js file for Joi validation or at least an if statement that checks if they exist